http2_server.py

```diff
@@ -0,0 +1,106 @@
+#!/usr/bin/env python
+import socket
+import email.utils
+import mimetypes
+import os
+
+
+ROOT = os.path.join(os.getcwd(), 'webroot')
+
+
+def response_ok(uri):
+    # send URI to be resolved
+    # include appropriate header reflecting correct content
+    # add body from appropriate resource
+
+    resolved_uri = resolve_uri(uri)
+    if resolved_uri[0] == 404:
+        response_code = "HTTP/1.1 404 Not Found"
+        headers = "Content-Type: text/plain"
+    else:
+        response_code = "HTTP/1.1 200 OK"
+        headers = "Content-Type: {}".format(resolved_uri[0])
+
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    response = "{}\r\n{}\r\n{}\r\n\r\n{}".format(
+        response_code, date, headers, resolved_uri[1])
+    return response
+
+
+def response_error(code, reason):
+    response_code = "HTTP/1.1 {} {}".format(code, reason)
+    date = "Date: {}".format(email.utils.formatdate(usegmt=True))
+    headers = "Content-Type: Text/HTML"
+    response = u"{}\r\n{}\r\n{}\r\n\r\n".format(response_code, date, headers)
+    return response
+
+
+def parse_request(request):
+    request = request.split('\r\n')
+    first_line = request[0].split(' ')
+    if first_line[0] != "GET":
+        return response_error(405, "Method Not Allowed")
+    elif first_line[2] != "HTTP/1.1":
+        return response_error(505, "HTTP Version Not Supported")
+    else:
+        return response_ok(first_line[1])
+
+
+def resolve_uri(uri):
+    # take URI from response_ok, return tuple of (content-type, body)
+    # if URI is directory, return listing of the directory (links)
+    # if URI not found, return 404
+    uri = uri.lstrip('/')
+    content = mimetypes.guess_type(uri)
```

Comment:
Using mime type's inability to guess the type of a directory is a terrible way to determine if something is a directory.  There's a much better way available to you in the os.path library.